### PR TITLE
Fix data race in prober test

### PR DESCRIPTION
**Pull request by timstclair**
Fixes https://github.com/kubernetes/kubernetes/issues/18440#issuecomment-163303291

### CODE REVIEW

File: pkg/kubelet/prober/manager.go [hunk 1]
```diff
@@ -79,20 +79,34 @@ func NewManager(
 	runner kubecontainer.ContainerCommandRunner,
 	refManager *kubecontainer.RefManager,
 	recorder record.EventRecorder) Manager {
+	m := newManager(statusManager, livenessManager, runner, refManager, recorder)
+	m.start()
```
Comment (yujuhong) [inline pkg/kubelet/prober/manager.go]: Can we expose the start() method and have kubelet calls it. This makes it consistent with how we start other managers and avoid creating redundant goroutines during tests.

Comment (timstclair) [inline pkg/kubelet/prober/manager.go]: Done.

**Comment by yujuhong** [comment]
Thanks!

**Comment by wojtek-t** [comment]
great - thanks!

**Comment by wojtek-t** [comment]
@k8s-bot e2e test this please

**Pull request merged**